portfolio/src/main/java/com/google/sps/servlets/TitanicDataServlet.java

```diff
@@ -0,0 +1,72 @@
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Returns titanic survivor data by class in a hash map*/
+@WebServlet("/titanic-data")
+public class TitanicDataServlet extends HttpServlet {
+
+  private LinkedHashMap<String, Double> titanicSurvivors = new LinkedHashMap<>();
+
+  @Override
+  public void init() {
+    Double firstPercent = 0.0;
+    Double secondPercent = 0.0;
+    Double thirdPercent = 0.0;
+
+    Double firstTotal = 0.0;
+    Double secondTotal = 0.0;
+    Double thirdTotal = 0.0;
+
+    Double firstSurvived = 0.0;
+    Double secondSurvived = 0.0;
+    Double thirdSurvived = 0.0;
+
+    Scanner scanner = new Scanner(getServletContext().getResourceAsStream(
+        "/WEB-INF/train.csv"));
+    while (scanner.hasNextLine()) {
+      String line = scanner.nextLine();
+      String[] cells = line.split(",");
+
+      Integer survived = Integer.valueOf(cells[0]);
+      Integer shipClass = Integer.valueOf(cells[1]);
+
+      switch(shipClass) {
+          case 1: 
+            firstSurvived += survived;
+            firstTotal += 1;
+          case 2: 
+            secondSurvived += survived;
+            secondTotal += 1;
+          case 3: 
+            thirdSurvived += survived;
+            thirdTotal += 1;
+      }
+
+      firstPercent = firstSurvived / firstTotal;
+      secondPercent = secondSurvived / secondTotal;
+      thirdPercent = thirdSurvived / thirdTotal;
+
+      titanicSurvivors.put("First Class", firstPercent);
+      titanicSurvivors.put("Second Class", secondPercent);
+      titanicSurvivors.put("Third Class", thirdPercent);
```

Comment:
I think there may be a better way to represent this data.  What you really want is this relationship:
class: passengers who survived, total passengers

As you're scanning through the file, instead of modifying local variables for each class, could you instead modify a class that represents these stats?  For example, imagine you have this class:

```
class ClassStats {
  Integer total;
  Integer survived;

  ClassStats() {
    this.total = 0;
    this.survived = 0;
  }

  void addPassenger(Integer passengerSurvived) {
    this.total += 1;
    this.survived += passengerSurvived;
  }

  Double getSurvivalRate() {
    return this.survived / this.total;
  }
}
```

You could instead have ClassStats firstClass = new ClassStats();
and then when the line is referencing a first class passenger, you just call firstClass.addPassenger(survived).

Then you could easily get the survival rate by calling firstClass.getSurvivalRate(). You could even think about creating a class for the entire PassengerRoster that has ClassStats for each class.

Some food for thought on how to better organize this data you're extracting.